src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -885,40 +1050,46 @@ void func_809C2F34(EnAob01* this, PlayState* play) {
     this->actor.prevPos = this->actor.world.pos;
 }
 
-void func_809C2FA0(void) {
+/**
+ * Generates a randomized list of dog text offsets and stores them in weekEventRegs.
+ * EnDg uses these text offsets to determine what text to show when the player picks
+ * the dog up while equipped with the Mask of Truth, and EnRacedog uses them to
+ * determine what condition the dog is in.
+ */
+void EnAob01_InitializeDogTextOffsets(void) {
     u8 i;
     u8 rand;
-    u8 index;
-    u8 orig;
-    u8 orig2;
-    u8 sp44[7];
-    u8 sp34[] = {
+    u8 dogTextOffsetsTemp;
+    u8 packedDogTextOffsets[7];
+    u8 dogTextOffsets[] = {
         0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,
     };
 
-    for (i = 0; i < ARRAY_COUNT(sp34); i++) {
+    // Swap around the dog text offsets randomly 14 times.
+    for (i = 0; i < ARRAY_COUNT(dogTextOffsets); i++) {
         rand = Rand_ZeroFloat(14.0f);
-        orig = sp34[i];
 
-        sp34[i] = sp34[rand];
-        sp34[rand] = orig;
+        dogTextOffsetsTemp = dogTextOffsets[i];
+        dogTextOffsets[i] = dogTextOffsets[rand];
+        dogTextOffsets[rand] = dogTextOffsetsTemp;
     }
 
-    for (i = 0; i < ARRAY_COUNT(sp44); i++) {
-        gSaveContext.save.saveInfo.weekEventReg[42 + i] = 0;
-        sp44[i] = 0;
+    // Clear out all weekEventRegs related to the dog text offsets.
+    for (i = 0; i < ARRAY_COUNT(packedDogTextOffsets); i++) {
+        CLEAR_WEEKEVENTREG_DOG_RACE_TEXT_OFFSETS(i);
+        packedDogTextOffsets[i] = 0;
     }
```

Comment:
And would need to add a comment to the WEEKEVENTREG's (and the define of `RACEDOG_COUNT`) saying the number of WEEKEVENTREG's needs to be in sync with `RACEDOG_COUNT`? (sorry this is really just 1 long comment spread out over many comments)